### PR TITLE
profiler: Add cgroup name to the process metadata

### PR DESCRIPTION
**Pull request by javierhonduco**
## Test plan

**Added tests**:

```
$ make test/profiler
--- PASS: TestGetValueAndDeleteBatchExactElements (0.00s)
=== RUN   TestThing
=== RUN   TestThing/single_cgroup_path
=== RUN   TestThing/single_cgroup_path_without_trailing_newline
=== RUN   TestThing/deeper_cgroup_path
=== RUN   TestThing/root_cgroup_path
=== RUN   TestThing/extract_cpu_controller_from_multiple_cgroup_controllers
=== RUN   TestThing/extract_cpu_controller_from_multiple_cgroup_controllers#01
=== RUN   TestThing/malformed_path_does_not_panic
=== RUN   TestThing/empty_path_does_not_panic
--- PASS: TestThing (0.00s)
    --- PASS: TestThing/single_cgroup_path (0.00s)
    --- PASS: TestThing/single_cgroup_path_without_trailing_newline (0.00s)
    --- PASS: TestThing/deeper_cgroup_path (0.00s)
    --- PASS: TestThing/root_cgroup_path (0.00s)
    --- PASS: TestThing/extract_cpu_controller_from_multiple_cgroup_controllers (0.00s)
    --- PASS: TestThing/extract_cpu_controller_from_multiple_cgroup_controllers#01 (0.00s)
    --- PASS: TestThing/malformed_path_does_not_panic (0.00s)
    --- PASS: TestThing/empty_path_does_not_panic (0.00s)
PASS
ok  	github.com/parca-dev/parca-agent/pkg/profiler	(cached)
```
**Bare metal**
image

**K8s**
image

Signed-off-by: Francisco Javier Honduvilla Coto <javierhonduco@gmail.com>

### CODE REVIEW

File: pkg/profiler/utils.go [hunk 1]
```diff
@@ -0,0 +1,69 @@
+// Copyright (c) 2022 The Parca Authors
```
Comment (kakkoyun) [inline pkg/profiler/utils.go:1]: I think we should move this under `pkg` to a metadata package. We have a `pkg/discovery` which is similar. 

In the same fashion, we have packages for parsing `kallsym` and process and perf maps.

**Review by kakkoyun — CHANGES_REQUESTED** [review]
Looking good. Just a nit about package organization.

**Comment by javierhonduco** [comment]
Addressed feedback

**Review by kakkoyun — APPROVED** [review]

**Comment by javierhonduco** [comment]
Add full cgroup path + rebase

**Pull request merged**